library/src/main/java/com/getstream/sdk/chat/adapter/ListenerContainer.kt

```diff
@@ -0,0 +1,109 @@
+package com.getstream.sdk.chat.adapter
+
+import com.getstream.sdk.chat.view.MessageListView.AttachmentClickListener
+import com.getstream.sdk.chat.view.MessageListView.GiphySendListener
+import com.getstream.sdk.chat.view.MessageListView.MessageClickListener
+import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListener
+import com.getstream.sdk.chat.view.MessageListView.ReactionViewClickListener
+import com.getstream.sdk.chat.view.MessageListView.ReadStateClickListener
+import com.getstream.sdk.chat.view.MessageListView.UserClickListener
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
+
+class ListenerContainer(
```

Comment:
I am just thinking if we shouldn't make this class `open` to allow customization and annotate its constructor with @JvmOverloads. WDYT?

Reply:
Until we have a use case for this, I'd like to avoid customizing this class. It would be hard to manage the lifecycle of custom instances of it (since it's really important that we only have a single one, owned by `MessageListView`). In fact, now that I think about it, I'd make this `internal`, and only expose it to the factory through a public interface, because clients overriding the factory might need to reference the listeners.

---

As for `@JvmOverloads`, since there's no clear order or priority between the listeners here, I don't think it would help a whole lot. I don't see it likely that you'd only provide the first couple of them specifically.